ai_powered_qa/custom_plugins/website_explorer.py

```diff
@@ -0,0 +1,92 @@
+from openai import OpenAI
+from pydantic import PrivateAttr
+
+from ai_powered_qa.components.plugin import Plugin, tool
+
+
+class WebsiteExplorer(Plugin):
+    name: str = "WebsiteExplorer"
+    _client: OpenAI = PrivateAttr()
+
+    def __init__(self, **data):
+        super().__init__(**data)
+        self._client = OpenAI()
+
+    @tool
+    def find_element_to_perform_action(self, action_description: str, html: str):
```

Comment:
Interesting concept. Is the idea that the agent only passes a snippet of the overall HTML, which is a good candidate for the selector? I think we might get even more out of it if the agent only receives the description of the page, and this plugin gets the HTML through a side channel.
That way, the agent only has the task of figuring out the suitable action in plain english, and isn't distracted by the cognitive load of the HTML. And the tool costs less token, as we would only be passing the decription of the action I want to take, and not the HTML.